phpBB/includes/acp/acp_extensions.php

```diff
@@ -65,22 +77,22 @@ function main()
 		* @var	string	ext_name		Extension name from request
 		* @var	int		safe_time_limit	Safe limit of execution time
 		* @var	int		start_time		Start time
+		* @var	string	tpl_name		Template file to load; leave empty to continue execution, filled in if ready to finish
 		* @since 3.1.11-RC1
 		*/
 		$u_action = $this->u_action;
-		$vars = array('action', 'u_action', 'ext_name', 'safe_time_limit', 'start_time');
+		$tpl_name = '';
+		$vars = array('action', 'u_action', 'ext_name', 'safe_time_limit', 'start_time', 'tpl_name');
 		extract($this->phpbb_dispatcher->trigger_event('core.acp_extensions_run_action', compact($vars)));
 
-		// Cancel action
-		if ($request->is_set_post('cancel'))
-		{
-			$action = 'list';
-			$ext_name = '';
-		}
+		// In case they have been updated by the event
+		$this->u_action = $u_action;
+		$this->tpl_name = $tpl_name;
 
-		if (in_array($action, array('enable', 'disable', 'delete_data')) && !check_link_hash($request->variable('hash', ''), $action . '.' . $ext_name))
+		// If tpl_name was set by the prior event, we are done
```

Comment:
That is the whole intention of this new event var, to be able to tell how to move on to the next step.  If you look closely, the last thing that is done in all the cases is setting the tpl_name and template vars for the followon action.
With the current situation, whatever you do in the event, you will come back AND execute the main switch, whatever is the action (default is the same as 'list').  So, the only way to tell the method that it has finished is to do something like this.  Otherwise, the only way to produce the same effect would be to call a "refresh" directly from the event, not too nice in my point of view.
But if you propose an alternative as to how to completely avoid the execution of the main switch (set "default" switch action to break, for example, instead of 'list', or create a new action called 'noaction', or whatever), I would be fine as well.
To be honest, I do not need this in the extension I am planning, but might be useful for others...